zipline/errors.py

```diff
@@ -191,6 +191,13 @@ class IncompatibleHistoryFrequency(ZiplineError):
 """.strip()
 
 
+class HistoryInInitialize(ZiplineError):
+    """
+    Raised when an algorithm calls history() in initialize.
+    """
+    msg = "history() can only be called in handle_data()"
```

Comment:
Is calling history in `before_trading_start` actually **supported**?  We've made sure it doesn't crash, but afaik there's basically no guarantees about its behavior, right?